### DISCUSSION TITLE
ring und matchid topic werden parallel gesendet

### DISCUSSION THREAD

**Discussion by alkertrazz** [frickelzeugs/FingerprintDoorbell#54] [Q&A]
Hallo zusammen,

zuerst Entschuldigung, dass ich einfach ein Issue eröffnet habe obwohl nicht unbedingt erforderlich war. Anscheinend wars schon zu spät für mich gestern.

Ich habe noch ein anderes Problem das ich nicht gelöst bekomme, in Home Assistant kommen immer beide Topic an. Unabhängig davon ob ich ein Ring oder ein matchID auslöse.

```
15:28:18 - Vor 32 Minuten - Spuren
Test mqtt topic fingerprintDoorbell/ring
15:28:18 - Vor 32 Minuten - Spuren
Neue Automatisierung mqtt topic fingerprintDoorbell/matchId
15:28:17 - Vor 32 Minuten - Spuren
Test mqtt topic fingerprintDoorbell/ring
15:28:17 - Vor 32 Minuten - Spuren
Neue Automatisierung mqtt topic fingerprintDoorbell/matchId
15:28:14 - Vor 32 Minuten - Spuren
Test mqtt topic fingerprintDoorbell/ring
15:28:14 - Vor 32 Minuten - Spuren
Neue Automatisierung mqtt topic fingerprintDoorbell/matchId
15:28:13 - Vor 32 Minuten - Spuren
Test mqtt topic fingerprintDoorbell/ring
15:28:13 - Vor 32 Minuten - Spuren
Neue Automatisierung mqtt topic fingerprintDoorbell/matchId
15:28:08 - Vor 32 Minuten - Spuren
Test mqtt topic fingerprintDoorbell/ring
15:28:08 - Vor 32 Minuten - Spuren
Neue Automatisierung  mqtt topic fingerprintDoorbell/matchId
15:28:05 - Vor 32 Minuten
Test  mqtt topic fingerprintDoorbell/ring
15:28:05 - Vor 32 Minuten
Neue Automatisierung  mqtt topic fingerprintDoorbell/matchId
15:28:01 - Vor 32 Minuten
Test  mqtt topic fingerprintDoorbell/ring
15:28:01 - Vor 32 Minuten
Neue Automatisierung  mqtt topic fingerprintDoorbell/matchId
15:27:58 - Vor 32 Minuten
Test  mqtt topic fingerprintDoorbell/ring
15:27:58 - Vor 32 Minuten
Neue Automatisierung  mqtt topic fingerprintDoorbell/matchId
```

nach meiner Beurteilung kann es nicht an überbrückten Lötstellen oder ähnliches liegen. Ich bin etwas ratlos gerade, das Protokoll des Fingerprintsensors zeigt alles richtig an

```
[2022-12-18 14:27:55 UTC]: No Match Found (Code 2)
[2022-12-18 14:27:58 UTC]: Match Found: 1 - E with confidence of 141
[2022-12-18 14:28:05 UTC]: Match Found: 1 - E with confidence of 237
[2022-12-18 14:28:13 UTC]: No Match Found (Code 2)
[2022-12-18 14:28:17 UTC]: No Match Found (Code 2)
```

Ich wäre sehr dankbar wenn Ihr mir einen kleinen Hinweis geben könntet woran es noch liegen könnte. Da ich nicht ganz so firm in der Thematik rund um MQTT bin.

Grüße
Patrick

**Comment by alkertrazz** [comment]
Nach diversen Lösungsversuchen, neuverlöten und einem 2. Esp32 Mini immer noch das gleiche Ergebnis. Es wird ein Ring und MatchId  parallel sgesendet. Wie wahrscheinlich ist es, dass der Fingerprintsensor beide Signale sendet? Ich verwende Home Assistent und Mosquitto Broker.

**Reply by frickelzeugs** [reply]
Hi, evtl. hab ich dein Problem falsch verstanden, aber werden bei dir nun wirklich immer ring und match gleichzeitig gesendet mit ring=on und match=[irgendeine Id > -1]? Oder schaust du evtl. nur auf die Topics und ignorierst die Values der Topics komplett? Bei einer Berührung des Sensors werden immer ALLE Topics aktualsiert, der Value ist das, worauf geschaut werden muss. Siehe dazu auch https://github.com/frickelzeugs/FingerprintDoorbell#configure-mqtt-connection

**Reply by alkertrazz** [reply]
Laut dem Protokoll des FP Sensors nur einmal, dennoch lösen beide automationen aus. Also die Values hatte ich tatsächlich nicht. Wo trage ich das denn ein? Unter Payload dann? Bei der MQTT Message habe ich zum Beispiel fingerprintDoorbell/matchId stehen das Feld payload steht leer.

**Reply by alkertrazz** [reply]
Stop, ich habe es verstanden. 

Ich bin davon ausgegangen, dass /matchId stellvertretend für alle Matches ist. Wenn ich das richtig verstanden habe muss ich für jeden Match eine eigene Automation anlegen?

es funktioniert jedenfalls. Vielen Dank für den Wink mit dem Zaunpfahl!
